security/DeputyGuardianModule-Proxy.md

```diff
@@ -0,0 +1,63 @@
+| Function Name | Description |
+| ---------------------- | ---------------------------------------------------------- |
+| setRespectedGameType() | Change the current `gameType` used by the `OptimismPortal` |
+| blacklistDisputeGame() | Blacklist a disputeGame into the `OptimismPortal` |
+| unpause() | Unpause the withdrawal of the Superchain |
+| pause() | Pause the withdrawal of the Superchain |
+| setAnchorState()| Set a state into `AnchorStateRegistry` |
+
+
+## Problem Statement
+
+The existing `DeputyGuardianModule` is not _proxified_. \
+This is inconvenient each time we want to upgrade the `DeputyGuardianModule` to add a new feature or to fix a potential bug. \
+As this is not a proxy this require to redeploy the `DeputyGuardianModule` on L1. And also, to update the `Guardian` contract to add the new `DeputyGuardianModule` address as authorized module. \
+Moreover, changing the `DeputyGuardianModule` will break the current PSPs setup as these pause transactions are presigned with the **previous** `DeputyGuardianModule`. \
+Thus, we need to generate new PSPs through the tedious process of a ceremony to add the new `DeputyGuardianModule` for making the PSPs valid again. \
+We are clearly seeing that this is not a sustainable solution for the long term. 
+
+Additionally, we have to simulate the new PSPs and share them with other member of the Superchain.
+
+## Proposed Solution
+
+![image](https://github.com/user-attachments/assets/099258f4-19fb-44b8-84e9-670b78a2f868)
+
+Here we propose some modification of the architecture with a proxy contract on top of the `DeputyGuardianModule`, this will allow us to upgrade the `DeputyGuardianModule` and keeping the same address.
+Thus, this will not invalidate the PSPs and we will not need to regenerate new PSPs each time we upgrade the `DeputyGuardianModule`.
+
+The owner of the proxy contract `DeputyGuardianModuleProxy` would be the `Security Council` as the `DeputyGuardianModule` is doing action into the `Guardian` contract that is owned by the `Security Council`.
+This will means that only the security council would be able to upgrade the implementation of the `DeputyGuardianModuleProxy`.
+
+In this solution, we propose to follow the current proxy standard used by the rest of the contracts. 
+
+## Alternatives Considered
+
+No other alternatives were considered for now.
+
+## Risks & Uncertainties
+
+### Security Model
+
+We don't see any particular risk with this current design for the long term as this is a common pattern to use a proxy contract for upgrading a contract.
+
+A potential issue can occur with the PSPs coverage.
+During the upgrade the PSPs are going to be invalid so we need to make sure to presign new PSPs with the new `DeputyGuardianModuleProxy` address and simulate the PSPs accordingly.
+
+Now, we are using a proxy executing the same actions than before will cost a more gas. We should ensure there no previous call that can run out of gas (really unlikely).
```

Comment:
What do you mean by "previous call that can run out of gas"?

Reply:
> previous call that can run out of gas?

First, this should **not** happen (this is extremely unlikely). 
But, as I don't know the full components used with the DGM, it is possible that some hardcoded call are made with a certain amount of gas to the DGM (with automated code for switching automatically to certain game type or automatically call `pause()` or `unpause()` like Automated Pause or automated game switcher).
Since the `delegatecall` will now be replacing the previous normal call into the DGM as this is a proxy this will be more expensive and can lead to DoS if there is **hardcoded gas limit** in some code that called the DGM methods automatically.